### PR TITLE
BFS Code in C

### LINKED ISSUE #755: Breadth First Search in C
I'm GSSoC participant.
I would like to work on Breadth First Search in C

### PR DESCRIPTION
**Pull request by susiejojo**
Fixes #755

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output. : handled in terms of input/output stream
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Uploading BFS code in C with adjacency matrix representation

#### Languages Used:
- C

#### Files Added:
- Breadth_FIrst_Search.c

### CODE REVIEW

File: Breadth_First_Search/Breadth_First_Search.c [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+		scanf("%d %d",&origin,&destin);
+		graph[origin-1][destin-1] = 1;
+	}
+}
+int main()
+{
+	graph_initiator();
+	BFS_run();
+	printf("\n");
+}
```
[the remark this answers is not in the document]
Comment (susiejojo) [reply Breadth_First_Search/Breadth_First_Search.c]: Done! The input/output file is in the new pull request though and the required changes have been made. Thanks so much for your suggestions!

**Comment by susiejojo** [comment]
Commit: 075cf92

File: Breadth_First_Search/Breadth_First_Search.c [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+//BFS FUNCTIONS:
+
+/*Runs BFS on every vertex of the graph that hasn't been visited(marked black or 1) yet */
+void BFS_run()	
+{
+    for (int i = 0; i < n; i++) // initialising state of each vertex as 0(white)
+		state[i] = 0;
+    int v;
+	for (int i = 0; i < n; i++)
```
Comment (jainaman224) [inline Breadth_First_Search/Breadth_First_Search.c]: Indentation fix.

File: Breadth_First_Search/Breadth_First_Search.c [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+void BFS(int node);
+
+
+//BFS FUNCTIONS:
+
+/*Runs BFS on every vertex of the graph that hasn't been visited(marked black or 1) yet */
+void BFS_run()	
+{
+    for (int i = 0; i < n; i++) // initialising state of each vertex as 0(white)
+		state[i] = 0;
```
Comment (jainaman224) [inline Breadth_First_Search/Breadth_First_Search.c]: Indentation needs to be fixed.

**Comment by jainaman224** [comment]
This file is tab indented. You can reindent with spaces in sublime to fix these issues.

**Comment by susiejojo** [comment]
Commit b468aab

**Comment by susiejojo** [comment]
Commit 002fd4e

**Pull request merged**